### PR TITLE
Issue #552: Plugin-distribute session auto-rename hook with opt-in via .wholework.yml

### LINKED ISSUE #552: auto-rename hook を plugin 配布化し .wholework.yml で opt-in 化
## 背景

Issue #549 で `/auto N` 実行時のセッション自動リネーム hook を実装した。現状はプロジェクト固有の `.claude/settings.json.template` に hook 設定があり、wholework リポジトリ専用。他リポジトリで使うには手動で hook 設定をコピーする必要があり、配布性が低い。

#549 の Non-Goals では「wholework plugin 配下に hook を入れる方式は不採用（plugin 利用者全員に強制適用される副作用が大きい）」としたが、これは **`.wholework.yml` による opt-in を考慮していなかった**ための判断。マーカーで明示的に opt-in する方式なら強制適用問題は解消し、配布性は plugin 経由が圧倒的に高くなる。

## 目的

hook を wholework plugin 配下に移し、`.wholework.yml` の `session-auto-rename: true` で opt-in する仕組みに変更する。これにより：

- plugin を入れるだけで配布完了（リポジトリごとの手動コピー不要）
- 既存 plugin 利用者には**強制適用されない**（default `false`）
- 有効化したいリポジトリは 1 行追加でリネーム機能 ON

### 変更内容

1. **`.wholework.yml` 新マーカー追加**
   - キー: `session-auto-rename`（boolean, default `false`）
   - 環境変数: `HAS_SESSION_AUTO_RENAME`

2. **`modules/detect-config-markers.md` の Marker Definition Table に登録**
   - 既存 `opportunistic-verify` / `skill-proposals` 等と同じパターンで 1 行追加

3. **`scripts/hook-rename-on-auto.sh` の冒頭で opt-in チェック追加**
   - `$CLAUDE_PROJECT_DIR/.wholework.yml` を読み、`session-auto-rename: true` を含まない場合は `exit 0`
   - 既存 hook ロジック（`/auto` パターン分岐、Issue title 取得、命名規則、truncate、JSON 出力）はそのまま

4. **plugin manifest (`.claude-plugin/plugin.json`) または plugin-level hooks 定義に `UserPromptSubmit` hook 登録**
   - command: `${CLAUDE_PLUGIN_ROOT}/scripts/hook-rename-on-auto.sh`
   - timeout: 5000ms
   - 登録形式（manifest 内 `hooks` field か separate `hooks.json` か）は spec で確定

5. **`docs/guide/customization.md` 更新**
   - 新マーカー `session-auto-rename` の説明追記
   - 既存 marker 一覧表に行追加

6. **bats テスト追加**
   - `.wholework.yml` 不在 → 空出力（既存挙動維持）
   - `session-auto-rename: false` → 空出力
   - `session-auto-rename: true` → 既存 hook ロジック発火（`/auto N` で JSON 出力）

### 既存 `.claude/settings.json.template` のエントリ

当面は残す（plugin hook と二重起動になるが、両方とも opt-in チェック後は同じ JSON を出力するので冪等。先に exit したほうの出力が使われ、後発は無害）。Plugin 経由配布が定着したら別 Issue で削除を検討。

### Pre-merge (auto-verified)

- [x]  `modules/detect-config-markers.md` の Marker Definition Table に `session-auto-rename` 行が追加されている
- [x]  `HAS_SESSION_AUTO_RENAME` 変数が定義されている
- [x]  hook script の冒頭で `.wholework.yml` の `session-auto-rename` キーを参照する opt-in チェックが実装されている
- [x]  wholework plugin の `hooks/hooks.json` に `UserPromptSubmit` hook エントリが追加されている（spec 確定: `plugin.json` ではなく `hooks/hooks.json` に登録）
- [x]  `hooks/hooks.json` の hook command が `hook-rename-on-auto.sh` を参照している
- [x]  `docs/guide/customization.md` に新マーカーの説明が追加されている
- [x]  CI で bats テストが pass（opt-in チェックの 3 パターンを含む）

### Post-merge

- [x] 別の git リポジトリ（wholework plugin を enable 済み）で `.wholework.yml` に `session-auto-rename: true` を追加して新セッション起動、`/auto N` でセッション名が変わることを確認 
- [x] `.wholework.yml` に `session-auto-rename` を書かない（または `false`）リポジトリで `/auto N` を打ち、セッション名が変わらないことを確認 

## Auto-Resolved Ambiguity Points

- **既存 `.claude/settings.json.template` エントリの扱い**: 当面残す（二重起動だが opt-in チェック後の冪等出力で無害）。Plugin 配布が定着後、別 Issue で削除を検討
- **plugin hook 登録の配置先**: `hooks/hooks.json`（plugin root）に決定（spec 確定）。Claude Code 公式ドキュメントにて、plugin hooks は `hooks/hooks.json` に記述する形式と確認。`plugin.json` に `hooks` フィールドは存在しない。

## Needs Refinement (defer to /spec)

- **plugin hook 登録の正式 schema 確定**: Claude Code plugin hooks の公式 schema (docs.anthropic.com) を参照して必須フィールド確認。`#551`（external-spec check に JSON schema 必須フィールド確認を追加）の手順を適用すべき。特に `hookEventName` 必須フィールド（#549 follow-up で判明）を含む完全な JSON 出力 schema の事前確認
- **`.wholework.yml` 読み込み方式**: hook script から直接 `grep '^session-auto-rename:' $CLAUDE_PROJECT_DIR/.wholework.yml` で読むか、`scripts/get-config-value.sh` 等の既存ヘルパを再利用するか。bash 3.2 互換維持
- **既存 plugin hooks の有無確認**: 既に `.claude-plugin/` 配下に hook 登録パターンがあれば踏襲、なければ #551 の手順に従い公式 docs で schema 確認

## 参照

- 派生元: #549（auto-rename hook 実装）の Non-Goals 再評価
- 関連: #551（spec の external-spec check 強化、plugin hook spec 参照時に適用）
- 既存マーカー実装例: `.wholework.yml` の `opportunistic-verify`, `skill-proposals` 等

### PR DESCRIPTION
**Pull request by saito**
## Summary

- `hooks/hooks.json` を新規作成し、plugin レベルの `UserPromptSubmit` hook として `hook-rename-on-auto.sh` を登録
- `scripts/hook-rename-on-auto.sh` の冒頭に opt-in チェックを追加（`CLAUDE_PROJECT_DIR/.wholework.yml` に `session-auto-rename: true` が設定されている場合のみ動作）
- `modules/detect-config-markers.md` に `session-auto-rename` / `HAS_SESSION_AUTO_RENAME` マーカーを追加
- `docs/guide/customization.md` と `docs/ja/guide/customization.md` に新マーカーの説明を追記
- `docs/structure.md` と `docs/ja/structure.md` に `hooks/` ディレクトリエントリを追加
- `tests/hook-rename-on-auto.bats` に opt-in 3 パターン（yml 不在、false、true）のテストを追加

## Verification (pre-merge)

- `modules/detect-config-markers.md` に `session-auto-rename` 行と `HAS_SESSION_AUTO_RENAME` 変数が追加されている ✅
- `scripts/hook-rename-on-auto.sh` の冒頭に opt-in チェックが実装されている ✅
- `hooks/hooks.json` に `UserPromptSubmit` hook エントリと `hook-rename-on-auto.sh` 参照が追加されている ✅
- `docs/guide/customization.md` に新マーカーの説明が追加されている ✅
- CI で bats テストが pass（opt-in チェックの 3 パターンを含む）

## Verification (post-merge)

- 別の git リポジトリ（wholework plugin を enable 済み）で `.wholework.yml` に `session-auto-rename: true` を追加して新セッション起動、`/auto N` でセッション名が変わることを確認
- `.wholework.yml` に `session-auto-rename` を書かない（または `false`）リポジトリで `/auto N` を打ち、セッション名が変わらないことを確認

closes #552

### CODE REVIEW

**Review by saito — COMMENTED** [review]
## 承認条件検証結果

| 条件 | 結果 | 備考 |
|------|------|------|
| `modules/detect-config-markers.md` に `session-auto-rename` 行が追加されている | PASS | — |
| `HAS_SESSION_AUTO_RENAME` 変数が定義されている | PASS | — |
| hook script の冒頭で `session-auto-rename` opt-in チェックが実装されている | PASS | — |
| `hooks/hooks.json` に `UserPromptSubmit` hook エントリが追加されている | PASS | — |
| `hooks/hooks.json` の hook command が `hook-rename-on-auto.sh` を参照している | PASS | — |
| `docs/guide/customization.md` に新マーカーの説明が追加されている | PASS | — |
| CI で bats テストが pass（opt-in チェックの 3 パターンを含む） | PASS | `Run bats tests` SUCCESS |

## CI ステータス

| ジョブ名 | ステータス | 備考 |
|---------|-----------|------|
| DCO | SUCCESS | — |
| Forbidden Expressions check | SUCCESS | — |
| Run bats tests | SUCCESS | — |
| Validate skill syntax | SUCCESS | — |
| macOS shell compatibility | SUCCESS | — |

## コードレビュー

### General Comments

なし（全 issue はファイル/行レベルで特定可能）

### Issue Count Summary

- line comments として投稿: 2 件（MUST: 0, SHOULD: 0, CONSIDER: 2）
- General Comments（body 内）: 0 件

---

次のアクション: `/merge 553` を実行してください。

File: tests/hook-rename-on-auto.bats [hunk 1]
```diff
@@ -144,3 +150,30 @@ MOCK
     OUTPUT=$(echo "$INPUT" | bash "$SCRIPT")
     [ -z "$OUTPUT" ]
 }
+
+@test "no .wholework.yml produces empty output" {
+    EMPTY_DIR="$BATS_TEST_TMPDIR/empty-proj"
+    mkdir -p "$EMPTY_DIR"
+    INPUT='{"prompt":"/auto 123"}'
+    OUTPUT=$(CLAUDE_PROJECT_DIR="$EMPTY_DIR" echo "$INPUT" | CLAUDE_PROJECT_DIR="$EMPTY_DIR" bash "$SCRIPT")
```
Comment (saito) [inline tests/hook-rename-on-auto.bats:158]: **[Edge Case] tests/hook-rename-on-auto.bats:158**

`CLAUDE_PROJECT_DIR="$EMPTY_DIR"` が `echo` コマンドの前にも付いていますが、`echo` はシェルビルトインで `CLAUDE_PROJECT_DIR` を参照しないため冗長です。他の2つのテスト（lines 167, 176）は正しく `echo "$INPUT" | CLAUDE_PROJECT_DIR="..." bash "$SCRIPT"` の形式を使用しています。

将来のリファクタリングで2つ目のプレフィックスを削除した場合、テストが `setup()` で設定されたデフォルトの `CLAUDE_PROJECT_DIR` に依存してしまう可能性があります。Severity: CONSIDER

**推奨修正:**
```bash
OUTPUT=$(echo "$INPUT" | CLAUDE_PROJECT_DIR="$EMPTY_DIR" bash "$SCRIPT")
```
（`echo` 前の `CLAUDE_PROJECT_DIR="$EMPTY_DIR"` プレフィックスを削除）

File: scripts/hook-rename-on-auto.sh [hunk 2]
```diff
@@ -4,6 +4,13 @@
 # Silent exit (empty output) on no match or gh failure to preserve existing session name.
 
 INPUT=$(cat)
+
+# Opt-in check: session-auto-rename must be true in .wholework.yml
+WHOLEWORK_YML="${CLAUDE_PROJECT_DIR:-}/.wholework.yml"
```
Comment (saito) [inline scripts/hook-rename-on-auto.sh:9]: **[Edge Case] scripts/hook-rename-on-auto.sh:9**

`${CLAUDE_PROJECT_DIR:-}` は `CLAUDE_PROJECT_DIR` が未設定の場合に `WHOLEWORK_YML` が `/.wholework.yml` に解決されます。動作上は安全ですが（ファイルが存在しないので `exit 0`）、spec では Claude Code が常に `CLAUDE_PROJECT_DIR` を設定すると明記されており、`:-` フォールバックが意図を曖昧にしています。Severity: CONSIDER

**推奨修正（オプション）:**
```bash
if [ -z "${CLAUDE_PROJECT_DIR:-}" ]; then exit 0; fi
WHOLEWORK_YML="${CLAUDE_PROJECT_DIR}/.wholework.yml"
```
空チェックを明示することで、前提条件が明確になります。

**Comment by saito** [comment]
## レビュー レスポンスサマリー


### Claude レビュー レスポンス

| アスペクト | ファイル | 行 | 指摘 | 対応 | Fix/Skip 理由 |
|-----------|---------|-----|------|------|--------------|
| Edge Case/Robustness | tests/hook-rename-on-auto.bats | 158 | echo 前の冗長な CLAUDE_PROJECT_DIR プレフィックス | Skip | CONSIDER レベル、動作に影響なし |
| Edge Case/Robustness | scripts/hook-rename-on-auto.sh | 9 | `${CLAUDE_PROJECT_DIR:-}` の意図曖昧さ | Skip | CONSIDER レベル、安全上の問題なし |

### 軽量再チェック結果

- Validate skill syntax: PASS（CI 確認済み）
- 新規 MUST 指摘: なし
- 承認条件更新: なし

### 承認条件サマリー

全 Pre-merge 条件 7/7 PASS（CI bats テスト含む）
